guides/cpp-intro.md

```diff
@@ -0,0 +1,225 @@
+
+# This code example
+
+This guide will use the following code example to explain how to use C++ with mbed.
+
+```cpp
+#include "mbed.h"
+
+DigitalOut myled(LED1);
+
+int main() {
+    while(1) {
+        myled = 1;
+        wait(0.4);
+        myled = 0;
+        wait(0.4);
+    }
+}
+```
+
+# Where general code is written
+
+When writing your code you write it inside the `int main()` function.
+This is the starting point for your program and every program will execute this function first.
+In our example the program will start by running the `while` loop.
+
+# Objects
+
+Objects in C++ works mostly like objects in other languages like Java.
+
+## Initialize object
+To intitalize an object you declare the object.
+
+```cpp
+DigitalOut myled(LED1);
+```
+
+This line creates a new object of type `DigitalOut` named `myled`.
+Writing that line will call the constructor of the object.
+In our case we are sending the constructor the parameter `LED1`.
+After this line we have the variable `myled` initialized with the parameter `LED1`.
+
+## Accessing attributes in objects
+
+To access an attribute on an object you use the dot notation.
+
+```cpp
+// access myAttribute of myObject
+myObject.myAttribute;
+
+int amount = myObject.count;
+
+// amount will now have the value of myObject.count
+printf("%d", amount);
+```
+
+## Calling functions on objects
+
+There is two ways of calling a function.
+If your object is a pointer you can call a function on an object use the arrow notation `->`.
+If your object is not a pointer you can use the dot notation `.`.
+An object can return a value or not.
+A value can be any value in C++ (int, char, object, struct, template....etc).
+
+```cpp
+MyObject *myObject;
+
+// add a value to our object
+myObject->setValue(0x12);
+
+// get the value from our object
+int value = myObject->getValue();
+
+// value will be 0x12
+printf("%x", value);
+
+MyObject otherObject;
+
+otherObject.setValue(0x12);
+
+int otherValue = otherObject.getValue();
+
+// otherValue will be 0x12
+printf("%x", otherValue);
+```
+
+## Operator overloading
+
+In C++ you may overload operators.
+This is very common when working with objects.
+Operator overloading means that you can define code that will say how an operator works with two objects.
+
+For instance, say we have the object `Point` that has an x and y coordinate and a constructor to initialize a point with two numbers.
+This object also has overloaded the `+` function which returns a new point with both values added together.
+
+```cpp
+Point a(2, 1);
+Point b(4, 2);
+
+Point c = a + b;
+
+// c will now have the values (6, 3)
+```
+
+All operators can be overloaded.
+You can see in our example code the line
+
+```cpp
+myled = 1
+```
+
+Here the `=` operator is overloaded and is used to set the value of the led.
+So when you change the value of `myled` here you are actually setting the value of the led.
+
+# Defines in C++
+
+A define is a way of changing values in C++.
```

Comment:
I wouldn't say define *changes* values. Maybe, 'stores constant values' or 'give names to constant values'